### PR TITLE
feat(search): supporting chinese glossaryterm full text retrieval(#3914)

### PR DESCRIPTION
**Pull request by Huyueeer**
## Checklist
- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [X] Links to related issues (if applicable)
- [X] Tests for the changes have been added/updated (if applicable)
- [X] Docs related to the changes have been added/updated (if applicable)

### Change
source of problem: #3914 
configure the analyzer to replace word segmentation in other languages supported by `main_tokenizer`, The tokenizer that has been tested includes `smartcn_tokenizer & ik_smart`, [Elasticsearch analysis plugins](https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis.html).

### CODE REVIEW

**Review by jjoyce0510 — COMMENTED** [review]
Wow - awesome PR! 

This looks great to me. Want another pair of eyes on it, then we can ship. (cc. @dexter-mh-lee)

Thank you @Huyueeer!

File: metadata-service/factories/src/main/resources/application.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ elasticsearch:
   username: ${ELASTICSEARCH_USERNAME:#{null}}
   password: ${ELASTICSEARCH_PASSWORD:#{null}}
   pathPrefix: ${ELASTICSEARCH_PATH_PREFIX:#{null}}
+  mainTokenizer: ${ELASTICSEARCH_MAIN_TOKENIZER:#{null}}
```
Comment (dexter-mh-lee) [inline metadata-service/factories/src/main/resources/application.yml:83]: nit: can we put this under index ? Been trying to put all index builder related variables into index

Comment (Huyueeer) [reply metadata-service/factories/src/main/resources/application.yml:83]: @dexter-mh-lee Of course, no problem.

Comment (shirshanka) [reply metadata-service/factories/src/main/resources/application.yml:83]: @Huyueeer When you get a chance to do this, we will merge it!

Comment (Huyueeer) [reply metadata-service/factories/src/main/resources/application.yml]: @dexter-mh-lee @shirshanka OK. I'm sorry for the delay. I've made changes.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Comment by xiangqiao123** [comment]
@Huyueeer

**Comment by Huyueeer** [comment]
> @Huyueeer In Chinese, two Chinese characters are very common. Can you help put MIN_LENGTH configurable? It will be very helpful for Chinese word segmentation
> 
> https://github.com/datahub-project/datahub/blob/72198f98c01a1aabe1043f33b9a16fa35061bf83/metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/SettingsBuilder.java#L211

@xiangqiao123 Sorry, this part should be rebuilt. It seems that you should find the person who implements this part

**Pull request merged**